torchaudio/utils/sox_utils.py

```diff
@@ -1,13 +1,27 @@
 from typing import List, Dict
+from functools import wraps
 
 import torch
 
-from torchaudio._internal import (
-    module_utils as _mod_utils,
-)
 
+def is_sox_available():
+    return torch.ops.torchaudio.is_sox_available()
 
-@_mod_utils.requires_module('torchaudio._torchaudio')
+
+def requires_sox():
+    if is_sox_available():
+        def decorator(func):
+            return func
+    else:
+        def decorator(func):
+            @wraps(func)
+            def wrapped(*args, **kwargs):
+                raise RuntimeError(f'{func.__module__}.{func.__name__} requires sox')
+            return wrapped
+    return decorator
```

Comment:
The implementation looks good, but can you put this in `torchaudio.extension` module?

`torchaudio.utils.sox_utils` are intended for configuring the libsox, such as buffer size, while `is_sox_available` is for extension management. so when we make other components selective and add other availability functions (such as for Kaldi and RNN transducer), we can put all the `is_XXX_available` in one place.

Reply:
Probably in `module_utils.py` is good.